packaging/markers.py

```diff
@@ -202,24 +202,6 @@ def _eval_op(lhs: str, op: Op, rhs: str) -> bool:
     return oper(lhs, rhs)
 
 
-class Undefined:
-    pass
-
-
-_undefined = Undefined()
-
-
-def _get_env(environment: Dict[str, str], name: str) -> str:
-    value: Union[str, Undefined] = environment.get(name, _undefined)
-
-    if isinstance(value, Undefined):
-        raise UndefinedEnvironmentName(
-            f"{name!r} does not exist in evaluation environment."
-        )
-
-    return value
```

Comment:
I this this still needs to exist because the user can still do something like

```python
Marker("asdsfd = 'whatever'").evaluate()
```

and this function raises a better exception.
```

Reply:
That'll fail in the parse state.

There's a better reason to keep this, we accept a legacy setuptools key in the parse stage that isn't populated in the default environment.

Reply:
@uranusjr that would raise `InvalidMarker` without triggering this code.
@pradyunsg `python_implementation` is aliased to `platform_python_implementation` so it won't trigger this code either.

```py
>>> import packaging.markers
>>> packaging.markers.Marker("platform_python_implementation == 'whatever'").evaluate()
False
```

Reply:
@pradyunsg @uranusjr Is there anything else about this hunk? I honestly don't see any way in which I can leave it, not call into internal code in tests, and have 100% coverage at the same time.

Reply:
If it’s not used we can drop it; it’s always easy to just add it back if needed anyway.